path_secrets.go

```diff
@@ -0,0 +1,94 @@
+package kubesecrets
+
+import (
+	"context"
+
+	"github.com/hashicorp/errwrap"
+	"github.com/hashicorp/vault/sdk/framework"
+	"github.com/hashicorp/vault/sdk/logical"
+)
+
+const secretsPrefix = "secret_name"
+
+func pathSecrets(b *secretsReaderBackend) *framework.Path {
+	return &framework.Path{
+		Pattern: framework.MatchAllRegex(secretsPrefix),
+
+		Fields: map[string]*framework.FieldSchema{
+			"secret_name": {
+				Type:        framework.TypeString,
+				Description: "Specifies the name of the kubernetes secret.",
+				Query:       true,
+				Required:    true,
+			},
+			"namespace": {
+				Type:        framework.TypeString,
+				Description: "Specifies the name of the kubernetes secret namespace.",
+				Query:       true,
+				Required:    true,
+			},
+		},
+
+		Callbacks: map[logical.Operation]framework.OperationFunc{
+			logical.ReadOperation: b.handleRead,
+		},
+		HelpDescription: pathInvalidHelp,
+		ExistenceCheck:  b.handleExistenceCheck,
+	}
+}
+
+func (b *secretsReaderBackend) handleRead(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	secretName := data.Get("secret_name").(string)
+	namespace := data.Get("namespace").(string)
+	b.Logger().Info("In handleRead() secretName: " + secretName + ", namespace: " + namespace)
+
+	if secretName == "" {
+		resp := logical.ErrorResponse("Missing secret name")
+		return resp, nil
+	}
+
+	if namespace == "" {
+		resp := logical.ErrorResponse("Missing secret namespace")
+		return resp, nil
+	}
+
+	fetchedData, err := b.KubeSecretReader.GetSecret(secretName, namespace, b.Logger())
+	if err != nil {
+		resp := logical.ErrorResponse("Error reading the secret data " + err.Error())
+		return resp, nil
+	}
+
+	// Generate the response
+	resp := &logical.Response{
+		Data: fetchedData,
+	}
+
+	return resp, nil
+}
+
+func (b *secretsReaderBackend) handleExistenceCheck(ctx context.Context, req *logical.Request, data *framework.FieldData) (bool, error) {
+	out, err := req.Storage.Get(ctx, req.Path)
```

Comment:
Again, just for understanding how this works: does it eventually get to handleRead and errors if handleRead returned an error?

Reply:
handleExistenceCheck is also a field in the Path: https://github.com/mesh-for-data/vault-plugin-secrets-kubernetes-reader/blob/1e8251eb63b9e1a03fb0df6115536a25944425da/path_secrets.go#L36

From Vault's code:
https://github.com/hashicorp/vault/blob/master/sdk/framework/path.go#L49

Reply:
I understand that. I meant what does req.Storage.Get do? How does it check if the secret exits (if that's what it's supposed to do). Just in case you know.

Reply:
That is actually leftover code. I will remove it. The check fro the secret is done in GetSecret : https://github.com/mesh-for-data/vault-plugin-secrets-kubernetes-reader/blob/1e8251eb63b9e1a03fb0df6115536a25944425da/kubernetes_secrets_reader.go#L27